src/main/java/com/couchbase/lite/replicator/Replication.java

```diff
@@ -406,20 +404,76 @@ public int getChangesCount() {
     /**
      * Update the lastError
      */
-    /* package */ void setLastError(Throwable lastError) {
+    protected void setLastError(Throwable lastError) {
         this.lastError = lastError;
     }
 
-    /* package */ String remoteCheckpointDocID() {
+    /**
+     * Following two methods for temporary methods instead of CBL_ReplicatorSettings implementation.
+     */
+    protected String remoteCheckpointDocID() {
         return replicationInternal.remoteCheckpointDocID();
     }
 
+    public Set<String> getPendingDocumentIDs(){
+        if(isPull() || (isRunning() && pendingDocIDs != null))
+            return pendingDocIDs;
```

Comment:
As the `pendingDocIDs` is reset on the work executor thread, to avoid race condition, I think accessing `pendingDocIDs` variable here should be in a synchronized block.